core/common/src/main/java/alluxio/worker/block/io/LocalFileBlockWriter.java

```diff
@@ -50,20 +49,17 @@ public LocalFileBlockWriter(String path) throws IOException {
   }
 
   @Override
-  public GatheringByteChannel getChannel() {
-    return mLocalFileChannel;
-  }
-
-  @Override
   public long append(ByteBuffer inputBuf) throws IOException {
     long bytesWritten = write(mLocalFileChannel.size(), inputBuf.duplicate());
     mPosition += bytesWritten;
     return bytesWritten;
   }
 
   @Override
-  public void transferFrom(ByteBuf buf) throws IOException {
-    mPosition += buf.readBytes(mLocalFileChannel, buf.readableBytes());
+  public long append(ByteBuf buf) throws IOException {
+    long bytesWritten = buf.readBytes(mLocalFileChannel, buf.readableBytes());
+    mPosition += bytesWritten;
+    return bytesWritten;
```

Comment:
Should we still name the api differently, since it is a very subtle difference in the argument which leads to very different behavior (copy or no copy)

Reply:
why different? both are no copy

Reply:
You're right, I'm fine with the naming then. As a side note, should we try to migrate to use `ByteBuf` instead of `ByteBuffer`? The main benefit would be not having to use `cleanDirectByteBuffer` and a consistent API so we don't need cases like this.

Reply:
SGTM